src/Modules/OrchardCore.Commerce/Services/PaymentService.cs

```diff
@@ -0,0 +1,167 @@
+            orderPart.ShippingAddress.Address,
+            orderPart.BillingAddress.Address);
+        if (cart?.Totals.Single() is not { } total) return null;
+
+        var checkoutShapes = (await _fieldsOnlyDisplayManager.DisplayFieldsAsync(
+                await _contentManager.NewAsync(Order),
+                "Checkout"))
+            .ToList();
+
+        var stripeApiSettings = (await _siteService.GetSiteSettingsAsync()).As<StripeApiSettings>();
+        var initPaymentIntent = new PaymentIntent();
+        if (!string.IsNullOrEmpty(stripeApiSettings.PublishableKey) &&
+            !string.IsNullOrEmpty(stripeApiSettings.SecretKey))
+        {
+            var paymentIntentId = _paymentIntentPersistence.Retrieve();
+            initPaymentIntent = await _stripePaymentService.InitializePaymentIntentAsync(paymentIntentId);
+        }
+
+        return new CheckoutViewModel
+        {
+            ShoppingCartId = shoppingCartId,
+            Regions = (await _regionService.GetAvailableRegionsAsync()).CreateSelectListOptions(),
+            OrderPart = orderPart,
+            SingleCurrencyTotal = total,
+            StripePublishableKey = (await _siteService.GetSiteSettingsAsync()).As<StripeApiSettings>().PublishableKey,
+            UserEmail = email,
+            CheckoutShapes = checkoutShapes,
+            PaymentIntentClientSecret = initPaymentIntent?.ClientSecret,
+        };
+    }
+
+    public async Task FinalModificationOfOrderAsync(ContentItem order)
+    {
+        // Saving addresses.
+        var userService = _userServiceLazy.Value;
+        var orderPart = order.As<OrderPart>();
+
+        if (_hca.HttpContext != null && await userService.GetFullUserAsync(_hca.HttpContext.User) is { } user)
+        {
+            var isSame = orderPart.BillingAndShippingAddressesMatch.Value;
+
+            await userService.AlterUserSettingAsync(user, UserAddresses, contentItem =>
+            {
+                var part = contentItem.ContainsKey(nameof(UserAddressesPart))
+                    ? contentItem[nameof(UserAddressesPart)].ToObject<UserAddressesPart>()!
+                    : new UserAddressesPart();
+
+                part.BillingAndShippingAddressesMatch.Value = isSame;
+                contentItem[nameof(UserAddressesPart)] = JToken.FromObject(part);
+                return contentItem;
+            });
+        }
+
+        order.DisplayText = T["Order {0}", order.As<OrderPart>().OrderId.Text];
+
+        await _contentManager.UpdateAsync(order);
+
+        if (_workflowManagers.FirstOrDefault() is { } workflowManager)
+        {
+            await workflowManager.TriggerEventAsync(nameof(OrderCreatedEvent), order, "Order-" + order.ContentItemId);
+        }
+
+        var currentShoppingCart = await _shoppingCartPersistence.RetrieveAsync();
+        currentShoppingCart?.Items?.Clear();
+
+        // Shopping cart ID is null by default currently.
+        await _shoppingCartPersistence.StoreAsync(currentShoppingCart);
```

Comment:
Isn't that just `StoreAsync(new ShoppingCart())`? Seems unnecessary to retrieve and clear the current cart.

Reply:
Maybe, although this is just the move of the already existing logic, so I wouldn't touch the logic itself.

Reply:
Let's note this somewhere, create an issue or something. Because I see that you don't want to experiment with anything, because you just moved the logic, so it could be risky. But we should note this.

Reply:
And link the issue here, if there is one.